runtime/wasm/src/module/stopwatch.rs

```diff
@@ -0,0 +1,66 @@
+// Copied from https://github.com/ellisonch/rust-stopwatch
+// Copyright (c) 2014 Chucky Ellison <cme at freefour.com> under MIT license
+
+use std::default::Default;
+use std::time::{Duration, Instant};
+
+#[derive(Clone, Copy)]
+pub struct TimeoutStopwatch {
+    /// The time the stopwatch was started last, if ever.
+    start_time: Option<Instant>,
+    /// The time the stopwatch was split last, if ever.
+    split_time: Option<Instant>,
+    /// The time elapsed while the stopwatch was running (between start() and stop()).
+    elapsed: Duration,
+}
+
+impl Default for TimeoutStopwatch {
+    fn default() -> TimeoutStopwatch {
+        TimeoutStopwatch {
+            start_time: None,
+            split_time: None,
+            elapsed: Duration::from_secs(0),
+        }
+    }
+}
+
+impl TimeoutStopwatch {
+    /// Returns a new stopwatch.
+    pub fn new() -> TimeoutStopwatch {
+        let sw: TimeoutStopwatch = Default::default();
+        return sw;
+    }
+
+    /// Returns a new stopwatch which will immediately be started.
+    pub fn start_new() -> TimeoutStopwatch {
+        let mut sw = TimeoutStopwatch::new();
+        sw.start();
+        return sw;
+    }
+
+    /// Starts the stopwatch.
+    pub fn start(&mut self) {
+        self.start_time = Some(Instant::now());
+    }
+
+    /// Stops the stopwatch.
+    pub fn stop(&mut self) {
+        self.elapsed = self.elapsed();
+        self.start_time = None;
+        self.split_time = None;
```

Comment:
I don't care, but as long as we're making changes note that `split_time` is unused. I'm surprised there is no warning for this.